### PR TITLE
Section 2.11: OOP

### PR DESCRIPTION
**Pull request by Michelle-Lohwt**
Although the code is from the same video, but I separate them into different files.

### CODE REVIEW

File: sec2_the-ruby-programming-language/2.11_oop_intro/6_initialize.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class Student
+  attr_accessor :first_name, :last_name, :email, :username, :password
+
+  def to_s
+    "First name: #{@first_name}, Last name: #{@last_name}, Username: #{@username}, 
+    Email: #{@email}, Password: #{@password}"
+  end
+
+  def initialize(fname, lname, uname, eml, pwd)
```
Comment (tlkiong) [inline sec2_the-ruby-programming-language/2.11_oop_intro/6_initialize.rb:11]: Always put `initialize` as the first method in the class.

File: sec2_the-ruby-programming-language/2.11_oop_intro/recap.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class User
+  attr_accessor :name, :email
+  def initialize(name, email)
```
Comment (tlkiong) [inline sec2_the-ruby-programming-language/2.11_oop_intro/recap.rb:3]: add some spacings yo

File: sec2_the-ruby-programming-language/2.11_oop_intro/3_getter.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# michelle.first_name("Michelle")
+# puts michelle                       # in `first_name': wrong number of arguments (given 1, expected 0) (ArgumentError)
+
+# Setter
+michelle.first_name = "Haha"
+puts michelle                         # First name: Haha
+
+# Getter
+# If need Getter, cannot have method
+puts michelle.first_name              # Haha
```
Comment (Michelle-Lohwt) [inline sec2_the-ruby-programming-language/2.11_oop_intro/3_getter.rb:43]: 1. Why when there is a getter, we cannot have the method (Refer Line 43, Line 34)
2. Is it possible to define setter without parameters? When usually do you need a setter without parameters?

File: sec2_the-ruby-programming-language/2.11_oop_intro/4_attr_setter_getter.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+michelle.last_name = "Loh"
+michelle.email = "michelle@example.com"
+
+puts michelle.first_name    # Michelle
+puts michelle.last_name     # Loh
+puts michelle.email         # michelle@example.com
+puts michelle.username      #                           //nothing is output
+
+# Why nothing is output in Line 33?
+# because we did not initialise the object but we initailise in the class
```
Comment (Michelle-Lohwt) [inline sec2_the-ruby-programming-language/2.11_oop_intro/4_attr_setter_getter.rb:36]: Refer Line 36, 15
When do we need to initialise in a class and why do we do that?

Comment (tlkiong) [reply sec2_the-ruby-programming-language/2.11_oop_intro/4_attr_setter_getter.rb:36]: wat do you mean by `initialise in a class`?

Comment (Michelle-Lohwt) [reply sec2_the-ruby-programming-language/2.11_oop_intro/4_attr_setter_getter.rb:36]: like this `@username = "michelle1"`

**Review by tlkiong — COMMENTED** [review]
1. What does setter do? It sets the value of a var in the class
2. What does a getter do? It gets the value of a var in the class

---

Based on the above 2 definition, it do not make sense to have a setter that doesn't take any params. You are not setting any way.  If you want to do this, its not call a setter.

`initialise in a class` is basically just giving a default value to the variable. For example, you have a class call `Dog`. And you want this object to have 4 legs. So, you would create the class like:

```
class Dog
  def initialize
    @num_of_legs = 4
  end
end
```

File: sec2_the-ruby-programming-language/2.11_oop_intro/3_getter.rb (hunk 3 above)
Comment (Michelle-Lohwt) [reply sec2_the-ruby-programming-language/2.11_oop_intro/3_getter.rb:43]: @tlkiong  how about this?

Comment (tlkiong) [reply sec2_the-ruby-programming-language/2.11_oop_intro/3_getter.rb:43]: @Michelle-Lohwt How about which? My comment below is to answer this as well.

**Review by tlkiong — APPROVED** [review]

**Pull request merged**